domains/transfer-agreements/TransferAgreements.API/API.UnitTests/Services/ProjectOriginWalletServiceTest.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Threading.Tasks;
+using API.Models;
+using API.Services;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using ProjectOrigin.Common.V1;
+using ProjectOrigin.WalletSystem.V1;
+using Xunit;
+
+namespace API.UnitTests;
+
+public class ProjectOriginWalletServiceTest
+{
+    private ProjectOriginWalletService service;
+    private WalletService.WalletServiceClient fakeWalletServiceClient;
+
+    public ProjectOriginWalletServiceTest()
+    {
+        var fakeLogger = Substitute.For<ILogger<ProjectOriginWalletService>>();
+        fakeWalletServiceClient = Substitute.For<WalletService.WalletServiceClient>();
+
+        service = new ProjectOriginWalletService(fakeLogger, fakeWalletServiceClient);
+    }
+    [Fact]
+    public async Task TransferCertificates_TransferAgreementNoEndDate_ShouldCallWalletTransferCertificate()
+    {
+        var transferAgreement = new TransferAgreement()
+        {
+            Id = Guid.NewGuid(), StartDate = DateTimeOffset.UtcNow, SenderName = "Producent A/S",
+            SenderTin = "32132112", ReceiverTin = "11223344"
+        };
+
+        var fakeGranularCertificatesResponse = CreateAsyncUnaryCall(
+            new QueryResponse { GranularCertificates = { CreateGranularCertificate(DateTimeOffset.UtcNow.AddDays(1), DateTimeOffset.UtcNow.AddDays(2))  } }
+        );
+
+
+        var fakeTransferResponse = CreateAsyncUnaryCall(
+            new TransferResponse() {}
+        );
+
+        SetupWalletServiceClient(fakeGranularCertificatesResponse, fakeTransferResponse);
+
+        await service.TransferCertificates(transferAgreement);
+
+        fakeWalletServiceClient.Received(1).TransferCertificateAsync(Arg.Any<TransferRequest>(), Arg.Any<Metadata>());
+    }
+
+    [Fact]
+    public async Task TransferCertificates_CertificateStartIsInvalidAndTANoEndDate_ShouldNotCallWalletTransferCertificate()
```

Comment:
Invalid makes you think that the start is set to something Invalid. Maybe something like _CertificateStartDateBeforeTAStartDateAnd...

Reply:
fixed in c68f1480